build.zig

```diff
@@ -9,18 +9,20 @@ pub fn build(b: *std.Build) anyerror!void {
     } });
 
     const kernel = b.addExecutable(.{
-        .root_source_file = b.path("src/kernel.zig"),
-        .optimize = optimize,
-        .target = target,
+        .root_module = b.addModule("kernel", .{
```

Comment:
Use `b.createModule` to not expose the `kernel` as a module from the buildsystem